### PR TITLE
Add a flag that allows run retries

### PR DESCRIPTION
**Pull request by ehankland**

### CODE REVIEW

File: perfkitbenchmarker/pkb.py [hunk 1]
```diff
@@ -376,12 +382,20 @@ def DoRunPhase(benchmark, name, spec, collector, timer):
   """
   deadline = time.time() + FLAGS.run_stage_time
   run_number = 0
+  consecutive_failures = 0
   while True:
+    samples = []
     logging.info('Running benchmark %s', name)
```
Comment (yuyantingzero) [inline perfkitbenchmarker/pkb.py]: Add logging about current retry count?

File: perfkitbenchmarker/pkb.py [hunk 2]
```diff
@@ -376,12 +382,20 @@ def DoRunPhase(benchmark, name, spec, collector, timer):
...
     events.before_phase.send(events.RUN_PHASE, benchmark_spec=spec)
     try:
       with timer.Measure('Benchmark Run'):
         samples = benchmark.Run(spec)
+    except Exception:
+      consecutive_failures += 1
+      if consecutive_failures > FLAGS.run_stage_retries:
+        raise
+    else:
+      consecutive_failures = 0
```
Comment (yuyantingzero) [inline perfkitbenchmarker/pkb.py]: why do you need line 397,398?

Comment (ehankland) [inline perfkitbenchmarker/pkb.py]: Because I only want to fail on consecutive failures not total failures

Comment (cwilkes) [inline perfkitbenchmarker/pkb.py]: the if/else whitespacing looks off.

Maybe put in the exception the reason why it failed, ie
```
raise Exception('Failed more then %d times' % (FLAGS.run_stage_retries, ))
```

Comment (ehankland) [inline perfkitbenchmarker/pkb.py]: The else matches the except (not the if). Sorry - hadn't pushed latest changes - the latest logs exceptions on every retry

File: perfkitbenchmarker/pkb.py (hunk 1 above)
Comment (ehankland) [inline perfkitbenchmarker/pkb.py]: Added logging on every failure below

**Pull request merged**